### PR TITLE
sql: alias CREATE INDEX ... INCLUDE with STORING

### LINKED ISSUE #46617: sql: add support for INCLUDE syntax
[Postgres added support for INCLUDE](https://www.postgresql.org/docs/12/sql-createindex.html) which behaves similarly to our STORING. Is it possible for us to make INCLUDE an alias for STORING? Or do we have to modify behavior for adding support?

### LINKED ISSUE #46618: sql: add unimplemented feature for INCLUDE
As discussed in https://github.com/cockroachdb/cockroach/issues/46617, Postgres has a syntax for INCLUDE. We should add unimplemented errors with links to GitHub for this feature. 
```SQL
CREATE UNIQUE INDEX rides_id ON rides (id) INCLUDE (rider_id);
invalid syntax: statement ignored: at or near "include": syntax error
SQLSTATE: 42601
DETAIL: source SQL:
CREATE UNIQUE INDEX rides_id ON rides (id) INCLUDE (rider_id)
```

### PR DESCRIPTION
**Pull request by otan**
Resolves https://github.com/cockroachdb/cockroach/issues/46617.
Resolves https://github.com/cockroachdb/cockroach/issues/46618.

Release note (sql change): Add support for using CREATE INDEX ....
INCLUDE (col1, col2, ...), which is an alias that PostgreSQL uses that
is analagous to our STORING (col1, col2, ...) syntax.

### CODE REVIEW

**Comment by rohany** [comment]
Overkill, but add a test and make sure that it actually stores the columns when you use include.
 
Other than that LGTM.

File: pkg/sql/parser/sql.y [hunk 1]
```diff
@@ -4284,14 +4284,14 @@ create_schema_stmt:
...
-//                            [USING HASH WITH BUCKET_COUNT = <shard_buckets>] [STORING ( <colnames...> )] [<interleave>]
+//                            [USING HASH WITH BUCKET_COUNT = <shard_buckets>] [{STORING | INCLUDE | COVERING} ( <colnames...> )] [<interleave>]
 //    FAMILY [<name>] ( <colnames...> )
 //    [CONSTRAINT <name>] <constraint>
 //
 // Table constraints:
 //    PRIMARY KEY ( <colnames...> ) [USING HASH WITH BUCKET_COUNT = <shard_buckets>]
 //    FOREIGN KEY ( <colnames...> ) REFERENCES <tablename> [( <colnames...> )] [ON DELETE {NO ACTION | RESTRICT}] [ON UPDATE {NO ACTION | RESTRICT}]
-//    UNIQUE ( <colnames... ) [STORING ( <colnames...> )] [<interleave>]
+//    UNIQUE ( <colnames... ) [{STORING | INCLUDE | COVERING} ( <colnames...> )] [<interleave>]
```
Comment (knz) [inline pkg/sql/parser/sql.y]: add the missing tests in `parse_test.go` too.

Comment (knz) [reply pkg/sql/parser/sql.y]: oh nvm you have it already. carry on!

**Comment by otan** [comment]
> Overkill, but add a test and make sure that it actually stores the columns when you use include.

hmm, what kind of test already tests this behaviour?

also i think logic tests will fail when using `INCLUDE` (or `COVERING`) since it double checks the pretty print matches the original text, but we don't encode this and since we alias it to STORING.

(maybe overkill indeed :P)

**Comment by rohany** [comment]
> also i think logic tests will fail when using INCLUDE (or COVERING) since it double checks the pretty print matches the original text, but we don't encode this and since we alias it to STORING.

oh true, this is pretty annoying. This is a fine amount of testing anyway.

**Comment by otan** [comment]
bors r=rohany

**Pull request merged**